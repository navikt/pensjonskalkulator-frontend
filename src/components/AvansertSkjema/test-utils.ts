export function getPreviousMonth(): string {
  const today = new Date()
  const currentMonth = parseInt(
    today.toLocaleDateString('en-CA').slice(5, 7),
    10
  ) // Extracts the month from the date string
  const previousMonth = currentMonth - 1
  return previousMonth.toString()
}
