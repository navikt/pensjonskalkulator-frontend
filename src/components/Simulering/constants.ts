export const MAX_UTTAKSALDER = 77
export const COLUMN_WIDTH = 25
export const TOOLTIP_YPOS = 35

export const SERIES_COLORS = {
  SERIE_COLOR_INNTEKT: 'var(--a-gray-500)',
  SERIE_COLOR_AFP: 'var(--a-purple-400)',
  SERIE_COLOR_TP: 'var(--a-green-400)',
  SERIE_COLOR_ALDERSPENSJON: 'var(--a-deepblue-500)',
  SERIE_COLOR_FADED_INNTEKT: 'var(--a-gray-300)',
  SERIE_COLOR_FADED_AFP: 'var(--a-purple-200)',
  SERIE_COLOR_FADED_TP: 'var(--a-green-200)',
  SERIE_COLOR_FADED_ALDERSPENSJON: 'var(--a-deepblue-200)',
}

export const SERIES_DEFAULT = {
  SERIE_INNTEKT: {
    type: 'column',
    pointWidth: COLUMN_WIDTH,
    name: 'Pensjonsgivende inntekt',
    color: SERIES_COLORS.SERIE_COLOR_INNTEKT,
  },
  SERIE_TP: {
    type: 'column',
    pointWidth: COLUMN_WIDTH,
    name: 'Pensjonsavtaler (arbeidsgivere m.m.)',
    color: SERIES_COLORS.SERIE_COLOR_TP,
  },
  SERIE_AFP: {
    type: 'column',
    pointWidth: COLUMN_WIDTH,
    name: 'AFP (Avtalefestet pensjon)',
    color: SERIES_COLORS.SERIE_COLOR_AFP,
  },
  SERIE_ALDERSPENSJON: {
    type: 'column',
    pointWidth: COLUMN_WIDTH,
    name: 'Alderspensjon (NAV)',
    color: SERIES_COLORS.SERIE_COLOR_ALDERSPENSJON,
  },
}

export const highchartsScrollingSelector = '.highcharts-scrolling'
