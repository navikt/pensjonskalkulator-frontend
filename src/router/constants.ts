export const BASE_PATH = '/pensjon/kalkulator'

export const externalUrls = {
  dinPensjon: 'https://nav.no/pensjon',
  dinPensjonBeholdning: 'https://www.nav.no/pensjon/opptjening/nb/',
  detaljertKalkulator: 'https://www.nav.no/pselv/simulering.jsf',
  alderspensjonsregler: 'https://www.nav.no/alderspensjon#beregning',
  afp: 'https://www.afp.no',
  afpPrivat: 'https://www.nav.no/afp-i-privat-sektor',
  garantipensjon: 'https://www.nav.no/minstepensjon',
  norskPensjon: 'https://norskpensjon.no/',
  uinnloggetKalkulator:
    'https://www.nav.no/pselv/simulering/forenkletsimulering.jsf',
  personvernerklaering:
    'https://www.nav.no/personvernerklaering#dine-rettigheter',
  personvernerklaeringKontaktOss:
    'https://www.nav.no/personvernerklaering#kontakt-nav',
  kontaktOss: 'https://www.nav.no/kontaktoss',
  planleggePensjon:
    'https://www.nav.no/planlegger-pensjon#noe-du-ikke-finner-svaret-p-her',
}

export const paths = {
  root: '/',
  login: '/login',
  henvisning: '/henvisning',
  start: '/start',
  samtykke: '/samtykke',
  utenlandsopphold: '/utenlandsopphold',
  offentligTp: '/offentlig-tp',
  afp: '/afp',
  samtykkeOffentligAFP: '/samtykke-offentlig-afp',
  ufoeretrygdAFP: '/ufoeretrygd-afp',
  sivilstand: '/sivilstand',
  uventetFeil: '/uventet-feil',
  beregningEnkel: '/beregning',
  beregningAvansert: '/beregning-detaljert',
  forbehold: '/forbehold',
  personopplysninger: '/personopplysninger',
} as const

export const henvisningUrlParams = {
  foedselsdato: '1963',
  apotekerne: 'apotekerne',
  utland: 'utland',
} as const
