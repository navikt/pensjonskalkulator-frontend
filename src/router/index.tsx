export { routes } from './routes'
export { externalUrls, paths, BASE_PATH } from './constants'
