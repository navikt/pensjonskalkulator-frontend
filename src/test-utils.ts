import renderer from 'react-test-renderer'

import { render } from '@testing-library/react'

export function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON()
  expect(result).toBeDefined()
  expect(result).not.toBeInstanceOf(Array)
  return result as renderer.ReactTestRendererJSON
}

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {
    // wrap provider(s) here if needed
    wrapper: ({ children }) => children,
    ...options,
  })

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
// override render export
export { customRender as render }
