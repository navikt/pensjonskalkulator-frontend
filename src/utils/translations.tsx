import { createIntl, createIntlCache } from 'react-intl'

import { ExternalLinkIcon } from '@navikt/aksel-icons'
import { Link } from '@navikt/ds-react'

import { getCookie, getTranslations } from '@/context/LanguageProvider/utils'
import { externalUrls } from '@/router/constants'

import { logOpenLink } from './logging'

/* c8 ignore next 1 */
const locale = getCookie('decorator-language') || 'nb'
const cache = createIntlCache()
const intl = createIntl(
  {
    locale,
    messages: getTranslations(locale),
  },
  cache
)

/* eslint-disable @typescript-eslint/no-explicit-any */
export const formatMessageValues: Record<string, any> = {
  detaljertKalkulatorLink: (chunks: string) => (
    <Link
      onClick={logOpenLink}
      href={externalUrls.detaljertKalkulator}
      target="_blank"
      inlineText
    >
      {chunks}
      <ExternalLinkIcon
        title={intl.formatMessage({ id: 'application.global.external_link' })}
        width="1.25rem"
        height="1.25rem"
      />
    </Link>
  ),
  dinPensjonLink: (chunks: string) => (
    <Link
      onClick={logOpenLink}
      href={externalUrls.dinPensjon}
      target="_blank"
      inlineText
    >
      {chunks}
      title={intl.formatMessage({ id: 'application.global.external_link' })}
      <ExternalLinkIcon
        title={intl.formatMessage({ id: 'application.global.external_link' })}
        width="1.25rem"
        height="1.25rem"
      />
    </Link>
  ),
  dinPensjonBeholdningLink: (chunks: string) => (
    <Link
      onClick={logOpenLink}
      href={externalUrls.dinPensjonBeholdning}
      target="_blank"
      inlineText
    >
      {chunks}
      <ExternalLinkIcon
        title={intl.formatMessage({ id: 'application.global.external_link' })}
        width="1.25rem"
        height="1.25rem"
      />
    </Link>
  ),
  alderspensjonsreglerLink: (chunks: string) => (
    <Link
      onClick={logOpenLink}
      href={externalUrls.alderspensjonsregler}
      target="_blank"
      inlineText
    >
      {chunks}
      <ExternalLinkIcon
        title={intl.formatMessage({ id: 'application.global.external_link' })}
        width="1.25rem"
        height="1.25rem"
      />
    </Link>
  ),
  garantiPensjonLink: (chunks: string) => (
    <Link
      onClick={logOpenLink}
      href={externalUrls.garantipensjon}
      target="_blank"
      inlineText
    >
      {chunks}
      <ExternalLinkIcon
        title={intl.formatMessage({ id: 'application.global.external_link' })}
        width="1.25rem"
        height="1.25rem"
      />
    </Link>
  ),
  afpLink: (chunks: string) => (
    <Link
      onClick={logOpenLink}
      href={externalUrls.afp}
      target="_blank"
      inlineText
    >
      {chunks}
      <ExternalLinkIcon
        title={intl.formatMessage({ id: 'application.global.external_link' })}
        width="1.25rem"
        height="1.25rem"
      />
    </Link>
  ),
  norskPensjonLink: (chunks: string) => (
    <Link
      onClick={logOpenLink}
      href={externalUrls.norskPensjon}
      target="_blank"
      inlineText
    >
      {chunks}
      <ExternalLinkIcon
        title={intl.formatMessage({ id: 'application.global.external_link' })}
        width="1.25rem"
        height="1.25rem"
      />
    </Link>
  ),
  navPersonvernerklaeringLink: (chunks: string) => (
    <Link
      onClick={logOpenLink}
      href={externalUrls.personvernerklaering}
      target="_blank"
      inlineText
    >
      {chunks}
      <ExternalLinkIcon
        title={intl.formatMessage({ id: 'application.global.external_link' })}
        width="1.25rem"
        height="1.25rem"
      />
    </Link>
  ),
  navPersonvernerklaeringKontaktOss: (chunks: string) => (
    <Link
      onClick={logOpenLink}
      href={externalUrls.personvernerklaeringKontaktOss}
      target="_blank"
      inlineText
    >
      {chunks}
      <ExternalLinkIcon
        title={intl.formatMessage({ id: 'application.global.external_link' })}
        width="1.25rem"
        height="1.25rem"
      />
    </Link>
  ),
  br: <br />,
}
/* eslint-enable @typescript-eslint/no-explicit-any */
