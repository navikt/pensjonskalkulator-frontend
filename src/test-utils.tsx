import React, { PropsWithChildren } from 'react'
import { IntlProvider } from 'react-intl'
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'

import { PreloadedState, createListenerMiddleware } from '@reduxjs/toolkit'
import { render, RenderOptions } from '@testing-library/react'

import { getTranslation_test } from '@/utils/__tests__/test-translations'

import { createUttaksalderListener } from './state/listeners/uttaksalderListener'
import {
  setupStore,
  RootState,
  AppStore,
  AppStartListening,
} from './state/store'
import { getTranslation_nb } from './translations/nb'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
  hasRouter?: boolean
}

export const swallowErrors = (testFn: () => void) => {
  const error = console.error
  console.error = () => {}
  testFn()
  console.error = error
}

export const swallowErrorsAsync = async (testFn: () => Promise<void>) => {
  const cache = console.error
  console.error = () => {}
  await testFn()
  console.error = cache
}

function generateMockedTranslations() {
  const nbTranslations: Record<string, string> = getTranslation_nb()
  const testTranslations: Record<string, string> = getTranslation_test()
  const translationsInput = { ...nbTranslations, ...testTranslations }
  const translations: Record<string, string> = {}

  for (const key in translationsInput) {
    if (
      /<(?=.*? .*?\/ ?>|br|hr|input|!--|wbr)[a-z]+.*?>|<([a-z]+).*?<\/\1>|{[^}]+}/i.test(
        translationsInput[key]
      )
    ) {
      // for html or chunks keys: results in 'my_key' : 'my_key with some <html> or {chunk}'
      translations[key] = translationsInput[key]
    } else {
      // results in 'my_key' : 'my_key'
      translations[key] = key
    }
  }
  return translations
}

// Return an object with the store and all of RTL's query functions
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState, true),
    hasRouter = true,
    ...renderOptions
  }: ExtendedRenderOptions = {}
) {
  function Wrapper({ children }: PropsWithChildren<unknown>): JSX.Element {
    return (
      <Provider store={store}>
        <IntlProvider locale="nb" messages={generateMockedTranslations()}>
          {hasRouter ? <MemoryRouter>{children}</MemoryRouter> : children}
        </IntlProvider>
      </Provider>
    )
  }

  const listenerMiddleware = createListenerMiddleware()
  createUttaksalderListener(
    listenerMiddleware.startListening as AppStartListening
  )

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'

export { renderWithProviders as render }
