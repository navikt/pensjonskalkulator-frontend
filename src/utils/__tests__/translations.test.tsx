import { FormattedMessage, IntlShape } from 'react-intl'

import { describe, expect, it } from 'vitest'

import { getFormatMessageValues } from '../translations'
import { externalUrls } from '@/router/constants'
import { render, screen } from '@/test-utils'

describe('translations-utils', () => {
  describe('formatMessageValues', async () => {
    const intlMock = {
      formatMessage: (s: { id: string }) => s.id,
    } as unknown as IntlShape

    it('formaterer <detaljertKalkulatorLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.detaljertKalkulatorLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()
      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()
      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        externalUrls.detaljertKalkulator
      )
      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <dinPensjonLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.dinPensjonLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()
      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()
      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        'https://nav.no/pensjon'
      )
      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <dinPensjonBeholdningLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.dinPensjonBeholdningLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()
      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()
      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        'https://www.nav.no/pensjon/opptjening/nb/'
      )
      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <dinPensjonEndreSoeknadLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.dinPensjonEndreSoeknadLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()
      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()
      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        'https://www.nav.no/pensjon/selvbetjening/alderspensjon/endringssoknad'
      )
      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <alderspensjonsreglerLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.alderspensjonsreglerLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()
      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()
      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        'https://www.nav.no/alderspensjon#beregning'
      )
      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <garantiPensjonLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.garantiPensjonLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()
      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()
      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        'https://www.nav.no/minstepensjon'
      )
      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <afpLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.afpLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()
      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()
      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        'https://www.afp.no'
      )
      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <afpPrivatLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.afpPrivatLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()
      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()
      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        'https://www.nav.no/afp-i-privat-sektor'
      )
      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <norskPensjonLink> med riktig url og uten ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.norskPensjonLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()
      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()
      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        'https://norskpensjon.no/'
      )
      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <navPersonvernerklaeringLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.navPersonvernerklaeringLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()

      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()

      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        externalUrls.personvernerklaering
      )

      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <navPersonvernerklaeringKontaktOssLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.navPersonvernerklaeringKontaktOssLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()

      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()

      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        externalUrls.personvernerklaeringKontaktOss
      )

      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <kontaktOssLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.kontaktOssLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()

      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()

      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        externalUrls.kontaktOss
      )

      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <planleggePensjonLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.planleggePensjonLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()

      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()

      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        externalUrls.planleggePensjon
      )

      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <trygdetidLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.trygdetidLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()

      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()

      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        externalUrls.trygdetid
      )

      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <kortBotidLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.kortBotidLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()

      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()

      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        externalUrls.kortBotid
      )

      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <personopplysningerLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.personopplysningerLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()

      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()

      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        externalUrls.personopplysninger
      )

      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer <spkLink> med riktig url og ikon', async () => {
      render(
        <FormattedMessage
          id="translation.test.spkLink"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(
        screen.getByText('lorem ipsum dolor', { exact: false })
      ).toBeInTheDocument()

      expect(screen.getByText('my link', { exact: false })).toBeInTheDocument()

      expect(screen.queryByRole('link')).toHaveAttribute(
        'href',
        externalUrls.personopplysninger
      )

      expect(
        await screen.findByRole('img', { hidden: true })
      ).toBeInTheDocument()
    })

    it('formaterer {br} til <br/>', async () => {
      const { asFragment } = render(
        <FormattedMessage
          id="translation.test.br"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(asFragment()).toMatchSnapshot()
    })

    it('formaterer <strong>', async () => {
      const { asFragment } = render(
        <FormattedMessage
          id="translation.test.strong"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(asFragment()).toMatchSnapshot()
    })

    it('formaterer <nowrap>', async () => {
      const { asFragment } = render(
        <FormattedMessage
          id="translation.test.nowrap"
          values={{ ...getFormatMessageValues(intlMock) }}
        />
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
