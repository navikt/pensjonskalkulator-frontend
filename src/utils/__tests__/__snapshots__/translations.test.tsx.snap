// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`translations-utils > formatMessageValues > formaterer <nowrap> 1`] = `
<DocumentFragment>
  lorem
  <span
    class="nowrap"
  >
    ipsum
  </span>
  dolor
</DocumentFragment>
`;

exports[`translations-utils > formatMessageValues > formaterer <strong> 1`] = `
<DocumentFragment>
  lorem
  <strong>
    ipsum
  </strong>
  dolor
</DocumentFragment>
`;

exports[`translations-utils > formatMessageValues > formaterer {br} til <br/> 1`] = `
<DocumentFragment>
  lorem
  <br />
  ipsum
  <br />
  <br />
  dolor
</DocumentFragment>
`;
